apis/Google.Cloud.Metadata.V1/Google.Cloud.Metadata.V1/MetadataClientImpl.cs

```diff
@@ -99,25 +99,125 @@ private static string GetETag(HttpResponseMessage response)
             return eTagValues?.FirstOrDefault();
         }
 
-        private static MaintenanceStatus GetMaintenanceStatus(string result)
+        private static T GetResult<T>(Func<CancellationToken, Task<T>> operation)
         {
-            switch (result)
+            try
+            {
+                return operation(CancellationToken.None).Result;
+            }
+            catch (AggregateException e)
+            {
+                throw e.InnerExceptions.FirstOrDefault() ?? e;
+            }
+        }
+
+        private static Instance ParseInstanceMetadata(string content)
```

Comment:
Is this the same code as there was before? It looks at least very similar, just moved - but I can't tell for sure.